spec/core/urlGeneration.ts

```diff
@@ -91,6 +91,28 @@ cases.push({
 		.query("$search=senior"),
 });
 
+cases.push({
+	url: "https://graph.microsoft.com/beta/me/people?$select=displayName,title,id&$count=false&$expand=a($expand=a,b)",
+	request: client
+		.api("/me/people")
+		.version("beta")
+		.select(["displayName", "title"])
+		.count(true)
+		.expand("a($expand=a,b)")
+		.query("$select=id")
+		.query("$count=false"),
+});
+
+cases.push({
+	url: "https://graph.microsoft.com/v1.0/me/people?$select=displayName,title,id&select=value",
+	request: client
+		.api("/me/people")
```

Comment:
Let's add cases where nested queries are set in `.api()` as that will exercise `parsePath`.

Reply:
Added. Please verify.

Reply:
Added more test cases. Please verify. The urlParsing.ts file has more test cases which apply to the parsePath() function.